src/biogeophys/OzoneMod.F90

```diff
@@ -565,4 +640,94 @@ subroutine CalcOzoneStressLombardozzi2015OnePoint( &
 
   end subroutine CalcOzoneStressLombardozzi2015OnePoint
 
+  
+  !-----------------------------------------------------------------------
+  subroutine CalcOzoneStressFalk(this, bounds, num_exposedvegp, filter_exposedvegp)
+    !
+    ! !DESCRIPTION:
+    ! Calculate ozone stress.
+    !
+    ! This subroutine uses the Falk formulation for ozone stress
+    !
+    ! !ARGUMENTS:
+    class(ozone_type), intent(inout) :: this
+    type(bounds_type), intent(in)    :: bounds
+    integer  , intent(in)            :: num_exposedvegp         ! number of points in filter_exposedvegp
+    integer  , intent(in)            :: filter_exposedvegp(:)   ! patch filter for non-snow-covered veg
+    !
+    ! !LOCAL VARIABLES:
+    integer  :: fp             ! filter index
+    integer  :: p              ! patch index
+
+    character(len=*), parameter :: subname = 'CalcOzoneStressFalk'
+    !-----------------------------------------------------------------------
+    
+    associate( &
+         o3uptakesha => this%o3uptakesha_patch                 , & ! Input:  [real(r8) (:)] ozone dose
+         o3uptakesun => this%o3uptakesun_patch                 , & ! Input:  [real(r8) (:)] ozone dose
+         o3coefjmaxsha => this%o3coefjmaxsha_patch             , & ! Output: [real(r8) (:)] ozone coef jmax sha
+         o3coefjmaxsun => this%o3coefjmaxsun_patch               & ! Output: [real(r8) (:)] ozone coef jmax sun
+         )
+      
```

Comment:
- [ ] [OPTIONAL] Since the coefficients set here are only needed at the end of the day, it might make sense to put an end-of-day conditional here, so this routine returns without doing anything on most time steps rather than doing unnecessary work. I'm a little torn on this, because I generally don't like having logic in one module that depends on logic in another module like this, but I'd lean towards adding this. (A better solution, which could make this dependency more obvious, could be to introduce a new function like is_time_to_run_luna, and replace the current checks of is_end_curr_day with a call to that function. This function would merely wrap is_end_curr_day, so the behavior would be the same, but the naming would give a clear signal to the code reader / maintainer about the reason we're doing this check – and then a change in the logic for when to run luna would happen in just that one place.)

Reply:
Thank you for the remainder of this @billsacks, I do have time for adding this change as well. 

To be 100% sure that I have understood correctly: 
- We introduce a public function called `is_time_to_run_luna` in `src/biogeophys/LunaMod.F90'. This is essentially a wrapper around the `is_end_curr_day`-function. 
- Inside `LunaMod.F90` we do the change `is_end_curr_day --> is_time_to_run_luna` 
- Inside `src/biogeophys/OzoneMod.F90` and the subroutine `CalcOzoneStressFalk` I introduce a conditional check so the current function body is only executed when `is_time_to_run_luna` is evaluated to true.

Reply:
Yes, that sounds right - thank you!